src/useAbort.ts

```diff
@@ -0,0 +1,25 @@
+import { useCallback, useState } from "react";
+
+export function useAbort() {
+  let abortController;
+
+  try {
+    abortController = new AbortController();
+  } catch {
+    abortController = null;
+  }
+
+  const [instance, setInstance] = useState(abortController);
```

Comment:
I would prefer to use useReference here, to avoid useless re-render 😉

Reply:
I did think that, but that means this hook has to expose the whole AbortController `instance`. If I'm not mistaken, if it's a reference and it gets a new `current` value set, the exposed `signal` won't get updated to the new one wherever it's consumed. The reason I wanted to exposed a wrapper like this is so the consuming code doesn't have to have checks everywhere to see if the instance is undefined.

Not sure if there's another/better solution? Or how I could in fact use `useRef` but still solve the problem in a similar way?

Reply:
@fabien0102 @TejasQ Could you shed any more light on this? 🙂 I'm not convinced the usage of `useState` is a big problem here since it would only cause a re-render if `abort` is called, and a re-render would also currently happen when the request resolves anyway. Would be interested to know if there is another solution though, and keen to be able to use this library with my Gatsby site!

Reply:
Logically you can expose what ever you want, nothing force you to expose the entire ref 😉
Not really sure of re-render here, I need a computer to try ^^

Can you tell me if the tests are passing, for a strange reason Travis seams blocked...

If yes, you can do/push a "npm version patch" so you will have a new npm package when we merge this PR 😉

Reply:
Not sure about `useState` usage, but you can improve code a bit like this

```ts
const [instance, setInstance] = useState(() => {
  try {
    return new AbortController();
  } catch {
    return null;
  }
});
```

Reply:
Apologies, it does look like the test for `useGet` are failing. I'll take a look and make the suggested change above as it's a lot cleaner 👍

Reply:
Don't worry, this is why we have some unit tests 😁